docs/memory-safety-continuum/memory-safety-continuum-draft2.md

```diff
@@ -0,0 +1,147 @@
+
+### 1. Writing software in Memory Safe By Default Languages
+
+Whenever possible/practical, you should use a memory safe by default language (such as Rust, Go, Python, Java, JavaScript, C#) when writing new software. Having memory safety incorporated directly into the language forces good safety practices at compile time as the code is being written. Developers get real time feedback as they create and compile their code. It is certainly possible and sometimes necessary to bypass some of the compiler's safety checks through the use of unsafe blocks and functions. However, memory safe by default languages force developers to intentionally choose unsafety (and, even when the do, to only use unsafety in contained and limited areas of the code), while non-memory safe by default languages depend on developers to intentionally (and continually) choose safety. If it is possible to escape into unsafe blocks, it is natural to ask whether developers will default to using them, negating the advantage of using a memory safe by default language. In 2022, as Google scaled up its use of Rust within the Android Open Source Project, they noted "In general, use of unsafe in Android’s Rust appears to be working as intended. It’s used rarely, and when it is used, it’s encapsulating behavior that’s easier to reason about and review for safety" [^1].
+
+Even when a language is memory safe by default, it is still vital to follow that language ecosystem's best practices and to use external tools to ensure safety not only within your code, but also within dependencies that your code pulls in. Nearly all Open Source software (and likely most close source software) depends on external libraries and packages to function. Care also needs to be exercised to check your dependencies for vulnerabilities.
+
+There are several ways to enhance the safety of your and your dependencies' code. These include:
+
+* Using the language ecosystem's best practices ([Examples](#memory-safe-by-default-language-ecosystem-best-practices))
+* Using automated tooling to provide additional safety checks to your code ([Examples](#memory-safe-by-default-language-automated-tooling-to-provide-additional-checks-to-your-code))
+* Using automated tooling to provide safety checks to your code's dependencies ([Examples](#memory-safe-by-default-language-automated-tooling-to-provide-additional-checks-to-your-dependencies))
+
+We have captured these enhancements in these substages of this section of the continuum, ordered from most to least ideal.
+
+* 1.1: Using developer best practices and automated tooling to verify safety in both your and your dependencies' code
```

Comment:
I find this numbered list a bit confusing after the bullet list above. do we need both or can we merge?

Reply:
Done!